config/squidGuard/squidguard_blacklist.php

```diff
@@ -58,11 +58,19 @@
 function squidguard_blacklist_AJAX_response( $request )
 {
     $res = '';
+    $status = '';
     $sz  = 0;
     $pcaption = '&nbsp;';
 
     # Actions
-    if     ($request['blacklist_download_start'])  squidguard_blacklist_update_start( $request['blacklist_url'] ); # update start
+    if ($request['blacklist_download_start'])
+        {
+                $url = $request['blacklist_url'];
+                if ( preg_match('/["\'\s\n\r\(\);]+/', $url )){
```

Comment:
A better test is to use:
`if (filter_var($url, FILTER_VALIDATE_URL) === false)`